app/utils/response_router.js

```diff
@@ -234,35 +239,47 @@ const actions = {
   }
 }
 
-exports.errorResponse = function errorReponse (req, res, reason, options, error) {
-  logger.error('Rendering error response', {
-    page: 'ERROR',
-    reason,
-    error,
-    ...getLoggingFields(req)
-  })
-  this.response(req, res, 'ERROR', options)
+exports.errorResponse = function errorReponse (req, res, reason, options = {}, error) {
+  const action = actions.ERROR
+  logErrorPageShown(action.view, reason, getLoggingFields(req), error)
+  options.viewName = 'ERROR'
+  redirectOrRender(req, res, actions.ERROR, options)
 }
 
-exports.systemErrorResponse = function systemErrorResponse (req, res, reason, options, error) {
-  logger.error('Rendering error response', {
-    page: 'SYSTEM_ERROR',
-    reason,
-    error,
-    ...getLoggingFields(req)
-  })
-  this.response(req, res, 'SYSTEM_ERROR', options)
+exports.systemErrorResponse = function systemErrorResponse (req, res, reason, options = {}, error) {
+  const action = actions.SYSTEM_ERROR
+  logErrorPageShown(action.view, reason, getLoggingFields(req), error)
+  options.viewName = 'SYSTEM_ERROR'
+  redirectOrRender(req, res, actions.SYSTEM_ERROR, options)
 }
 
 exports.response = function response (req, res, actionName, options) {
   options = options || {}
   options.viewName = actionName
   let action = lodash.result(actions, actionName)
+
+  if (action && action.logErrorPageShown) {
+    logErrorPageShown(action.view, `Action: ${actionName}`, getLoggingFields(req))
+  }
```

Comment:
Would it make sense for all of this logic to go in the `render` method itself? That way none of the higher level functions would have to call this logging directly. 

It could be removed from `repsonse`, `systemErrorResponse` and `errorResponse` and let the logic not branch in multiple places.